baybe/utils/plotting.py

```diff
@@ -0,0 +1,78 @@
+"""Plotting utilities."""
+
+import json
+import os
+import sys
+import warnings
+from pathlib import Path
+
+import matplotlib.pyplot as plt
+import pandas as pd
+import seaborn as sns
+
+
+def create_plots(data: pd.DataFrame, path: Path, base_name: str, **kwargs) -> None:
+    """Create plots from a given data frame and save them as a svg file.
+
+    The plots will be saved in the location specified by ``path``.
+    The attribute ``base_name`` is used to define the name of the outputs.
+
+    If the ``SMOKE_TEST`` variable is set, no plots are being created and this method
+    immediately returns.
+
+    Using the ``BAYBE_MULTIVERSION_PLOTS`` environment variable, it is possible to
+    create plots for the light and dark version of the documentation. If this variable
+    is not set, a single file named ``{base_name}_check.svg`` is created.
+
+    Args:
+        data: The data frame containing the data to be plotted.
+        path: The path to the directory in which the plots should be saved.
+        base_name: The base name that is used for naming the output files.
+        **kwargs: Keyword arguments. They are directly passed to ``sns.lineplot`` and
+            are used for specifying the plot.
+    """
+    # If we run a SMOKE_TEST, we immediately return
+    if "SMOKE_TEST" in os.environ:
+        return
+    # We see if we happen to find the plotting themes in the current folder.
+    # This is e.g. the case if we convert the file to a jupyter notebook.
+    try:
+        themes = json.load(open("plotting_themes.json"))
+    except FileNotFoundError:
+        # Try to find the plotting themes by backtracking
+        # Get the absolute path of the current script
+        script_path = Path(sys.argv[0]).resolve()
+        # Backtrack until either the "baybe" folder is found or backtracking is no
+        # longer possible
+        while not script_path.name == "baybe" and script_path != script_path.parent:
+            script_path = script_path.parent
+        if script_path == script_path.parent:
+            warnings.warn("No themes for plotting found. A default theme is used.")
+            themes = {"check": {"font_scale": 1.75, "rc_params": {}}}
+        else:
+            # Open the file containing all the themes
+            # This currently still assumes that the file is in the repo folder
+            themes = json.load(open(script_path / "plotting_themes.json"))
+
+    # Environment variables for checking whether we want to have multiversion plots
+    BAYBE_MULTIVERSION_PLOTS = "BAYBE_MULTIVERSION_PLOTS" in os.environ
+
+    # Choose either all available or just the `check` theme.
+    chosen_themes = themes if BAYBE_MULTIVERSION_PLOTS else ["check"]
```

Comment:
is this new env variable really necessary? ie why not just make all plots all the time?

Reply:
Idea was that this prevents us and users from overwriting the "nice" plots that are used for the visualization by accident. However, if we either decide that this is a risk we are willing to take, we could also get rid of them.

Reply:
since they are stored on github the local overwrite should be no problem

Reply:
Removed the variable.